File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0], maxProfit = 0;
        for(int i=1; i<prices.length; i++){
            min = Math.min(min, prices[i-1]);
            maxProfit = Math.max(maxProfit, prices[i] - min);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let maxAmountOfWater = 0;
        for(let i=0; i<heights.length; i++){
            for(let j=i+1; j<heights.length; j++){
                let currentMaxWater = Math.min(heights[i], heights[j]) * (j-i)
                if(maxAmountOfWater < currentMaxWater) maxAmountOfWater = currentMaxWater
            }
        }
        return maxAmountOfWater
    }
}
